extensions/libraries/redcore/api/api.php

```diff
@@ -199,9 +199,27 @@ public static function getPostedData()
 				}
 			}
 
-			// We try to transform it into XML
-			if (is_null($parsedData) && $xml = @simplexml_load_string($inputData))
+			$xml = @simplexml_load_string($inputData);
+
+			// We try to transform soap XML
+			if (is_null($parsedData) && $xml !== false)
 			{
+				/*
+					XML Namespaces suck and I couldn't get it to read the data accurately for SOAP
+					So I'm stripping everything between < and : in the xml and re-rendering it
+				    I know. I know. My skin is crawling too, but its the best I could do at the moment.
+				    If someone can make this work with namespaces that would be B-E-A-UTIFUL!
+				*/
+				$cleanData = preg_replace('/<(\/)?([a-z0-9]+):([a-z0-9]+)/i', '<$1$3', $inputData);
+				$xml = @simplexml_load_string($cleanData);
```

Comment:
sorry, you handled the code so well so I thought it was not needed :)
You can get language parameters from $data where you have actually that parameter in the webservice. Also only place where you need it is in read item and read list...
I have already created the function for setting the language (and you have seen it for sure :) ): https://github.com/redCOMPONENT-COM/redCORE/blob/develop/extensions/libraries/redcore/api/soap/operation/operation.php#L337 so please just set JFactory::getApplication()->input here and you should be good to go and wont have to mess with incomming parsing of xml envelope from soap request or anthing like that :)